model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/sphinx.py

```diff
@@ -0,0 +1,99 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Sphinx documentation plugin used to document decorators.
+
+Introduction
+============
+
+Usage
+-----
+
+The extension requires Sphinx 2.0 or later.
+
+Add the extension to your :file:`docs/conf.py` configuration module:
+
+.. code-block:: python
+
+    extensions = (...,
+                  'path.to.here.sphinx')
+
+With the extension installed `autodoc` will automatically find gt4py decorated
+objects with @field_operator, @scan_operator, @program (e.g. when using the
+automodule directive) and generate the correct (as well as add a
+``(gt4pydecor)`` prefix).
+"""
+import inspect
+
+from docutils import nodes
+from gt4py.next.ffront.decorator import FieldOperator, Program
+from gt4py.next.ffront.field_operator_ast import ScanOperator
+from sphinx.domains import python as sphinx_python
+from sphinx.ext import autodoc
+
+
+class Gt4pydecorDocumenter(autodoc.FunctionDocumenter):
+    """Document gt4pydecor definitions."""
+
+    objtype = "gt4pydecor"
+    member_order = 11
+
+    @classmethod
+    def can_document_member(cls, member, membername, isattr, parent):
+        return isinstance(member, (Program, FieldOperator, ScanOperator)) and member.__wrapped__
+
+    def format_args(self):
+        wrapped = getattr(self.object, "__wrapped__", None)
+        if wrapped is not None:
+            sig = inspect.signature(wrapped)
+            if "self" in sig.parameters or "cls" in sig.parameters:
```

Comment:
can you explain what you are trying to check here?

Reply:
@havogt?